mne/io/egi/source/general.py

```diff
@@ -0,0 +1,349 @@
+                segind = np.where(ep_begins == segbeginsamp)[0]
+                epochsegstatus[segind[0]] = aseg.getAttribute('status')
+                epochlabels[segind[0]] = categlabel
+                time0 = int(aseg.getElementsByTagName('evtBegin')[0]
+                            .firstChild.data)
+                time0samp = _u2sample(time0, samprate)[0]
+                time0samp = time0samp - segbeginsamp
+                epochtime0[segind[0]] = time0samp
+                if multisubj:
+                    keylistarray = aseg.getElementsByTagName('keyCode')
+                    numkeys = keylistarray.length
+                    datalistarray = aseg.getElementsByTagName('data')
+                    for r in range(numkeys):
+                        akey = keylistarray[r]
+                        adata = datalistarray[r]
+                        subject = None
+                        filename = None
+                        if adata.firstChild is not None:
+                            if akey.firstChild.data.encode() == 'subj':
+                                subject = adata.firstChild.data   # .encode()
+                            elif akey.firstChild.data.encode() == 'FILE':
+                                filename = adata.firstChild.data.encode()
+                        epochsubjects[segind[0]] = subject
+                        epochfilenames[segind[0]] = filename
+    l_epfilen = len(set(epochfilenames))
+    l_epsubjects = len(set(epochsubjects))
+    if (multisubj and l_epfilen == 1 and l_epsubjects == 1):
+        epochsubjects = []
+        epochfilenames = []
+        multisubj = False
+    #  --------------------------------------------------------------------
+    epoch_info = dict(epochType=epochtype,
+                      epochBeginSamps=ep_begins,
+                      epochNumSamps=epochnumsamps,
+                      epochFirstBlocks=epochfirstblocks,
+                      epochLastBlocks=epochlastblocks,
+                      epochLabels=epochlabels,
+                      epochTime0=epochtime0,
+                      multiSubj=multisubj,
+                      epochSubjects=epochsubjects,
+                      epochFilenames=epochfilenames,
+                      epochSegStatus=epochsegstatus)
+    return epoch_info
+
+
+def _get_signal_nbin(filepath, signalnbin):
+    import numpy as np
+
+    binfile = filepath + '/' + signalnbin
+    with open(binfile, 'rb') as fid:
+        s = fid.read()
+        l = len(s)
+        fid.seek(0, 0)
+        version = np.fromfile(fid, dtype=np.dtype('i4'), count=1)[0]
+        if version != 0:     # siempre el primer valor version es 1?
+            block = _block_r(fid)
+            blocksize = block['blocksize']
+            position = block['position']
+            nsamples = block['nsamples']
+        else:
+            raise NotImplementedError('Only continuos files are supported')
+        numblocks = int(l // blocksize)
```

Comment:
@ramonapariciog the problem was here. It was previously not doing integer division and then adding an extra block if the data did not align. If I understand correctly, the proper thing to do is to just dump the extra data, since it is only available for the first channels in the block.